src/Microsoft.OData.Core/UriParser/Binders/SelectExpandBinder.cs

```diff
@@ -894,6 +901,11 @@ private static LevelsClause ParseLevels(long? levelsOption, IEdmType sourceType,
                 state.RangeVariables.Push(explicitRangeVariable);
             }
 
+            // Create $this rangeVariable and add it to the Stack.
+            RangeVariable dollarThisRangeVariable = NodeFactory.CreateDollarThisRangeVariable(
+                elementType != null ? elementType : targetNavigationSource.EntityType().ToTypeReference(), targetNavigationSource);

```

Comment:
@KenitoInc I assume that we're guaranteed a non-null from this expression: `elementType != null ? elementType : targetNavigationSource.EntityType().ToTypeReference()`?

Reply:
As far as I know, Yes!